core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -206,6 +191,89 @@ public void testFailedCommit() throws Exception {
     Assertions.assertThat(manifests).as("Should contain 0 Avro manifest files").isEmpty();
   }
 
+  @Test
+  public void testCommitFailedBeforeChangeVersionHint() throws Exception {
+    // create v1-format table,for easy testing
+    Map<String,String> confMap = new HashMap<>();
+    confMap.put(TableProperties.FORMAT_VERSION,"1");
+    String newLocation = tableLocation+"v1"+File.separator;
+    Table v1Table = TABLES.create(SCHEMA, SPEC,confMap,newLocation);
+    // Submit a new file to test the functionality.
+    v1Table.newFastAppend().appendFile(FILE_A).commit();
+    BaseTable baseTable = (BaseTable) v1Table;
+    HadoopTableOperations tableOperations = (HadoopTableOperations) baseTable.operations();
+    HadoopTableOperations spyOps = spy(tableOperations);
+    doThrow(new RuntimeException("hdfs crash!"))
+            .when(spyOps)
+            .renameToFinal(any(), any(), any(), any());
+    int versionBefore = spyOps.findVersion();
+    TableMetadata metadataV1 = spyOps.current();
+    TableMetadata metadataV2 = metadataV1.upgradeToFormatVersion(2);
```

Comment:
Why are we updating the format version?

Reply:
The reason for this is simple: I just want to override the execution logic of the commit method, no matter what I do.

Reply:
I don't understand why the append then as well?

Reply:
you mean why i execute ```v1Table.newFastAppend().appendFile(FILE_A).commit();```  ?
It's just that I want to test that the commit function of the table I created works. I want to prove that the table I created is OK, and based on that, I'll run my real test case.